### ISSUE TITLE
About the results

### ISSUE THREAD

**Issue by Ge0sz** [ar4/deepwave#26]
Greetings, i can't understand the results from the receiver_amplitudes. I get a shape of (250,250,1) and the data shape is (201,301). My code is the following: 

`dx = 10 # 10m in each dimension
dt = 0.004 # 4ms --- Sampling Interval
nz = np.shape(data)[0]
ny = np.shape(data)[1]
nt = int(1 / dt) # 1s --- Sampling freq
r_spacing = dx*(ny-1)
s_spacing = dx
ymax = (ny-1)*dx
num_shots = int(np.floor(ymax / s_spacing))
coord_receivers = tr.linspace(0,ymax,2)
num_receivers_per_shot = int(len(coord_receivers))
num_sources_per_shot = 1
num_dims = 2
peak_freq = 4
peak_source_time = 1/peak_freq

#  model
model = tr.tensor(data)

"""
  source and receiver position --- x_s -> Tensor of shape [num_shots, num_sources_per_shot, num_dimensions]
                               --- x_r -> Tensor of shape [num_shots, num_receivers_per_shot, num_dimensions]
"""

x_r = tr.zeros(num_shots,num_receivers_per_shot,num_dims)
for j in range(num_receivers_per_shot):
  x_r[:,j,1] = coord_receivers[j] #receiver potition

x_s = tr.zeros(num_shots, num_sources_per_shot, num_dims)
for j in range(num_sources_per_shot):
  x_s[:,0,1] = tr.linspace(ymax/2, ymax, num_shots) #sources potition
  x_s[:,0,0] = 100#tr.linspace(0,(nz-1)*dx, len(x_s[:,0,1])) #depth
  

#  Ricker Wavelet
source_amplitudes = (dp.wavelets.ricker(peak_freq, nt, dt,
                                             peak_source_time).reshape(-1, 1, 1).repeat(1, num_shots, num_sources_per_shot))
#  
prop = dp.scalar.Propagator({'vp': model}, dx)
receiver_amplitudes = prop(source_amplitudes, x_s, x_r, dt)`

**Comment by ar4** [comment]
Greetings,

I ran your code and obtained a `receiver_amplitudes` of shape [250, 300, 2], as expected, and the result looked plausible. To be certain, I have pasted the code that I ran, and attached the output that I received. Does this code produce a `receiver_amplitudes` of shape [250, 300, 1] for you? If so, did you receive any warning messages while running it?
![receiver0](https://user-images.githubusercontent.com/10000682/112472013-830f5c00-8d64-11eb-9fb6-8416132d3e1e.jpg)
![receiver1](https://user-images.githubusercontent.com/10000682/112471661-13996c80-8d64-11eb-9aa3-90a5014fe94d.jpg)

```
import deepwave as dp
import numpy as np
import torch as tr

data = np.ones([201, 301], np.float32)*2500

dx = 10 # 10m in each dimension
dt = 0.004 # 4ms --- Sampling Interval
nz = np.shape(data)[0]
ny = np.shape(data)[1]
nt = int(1 / dt) # 1s --- Sampling freq
r_spacing = dx*(ny-1)
s_spacing = dx
ymax = (ny-1)*dx
num_shots = int(np.floor(ymax / s_spacing))
coord_receivers = tr.linspace(0,ymax,2)
num_receivers_per_shot = int(len(coord_receivers))
num_sources_per_shot = 1
num_dims = 2
peak_freq = 4
peak_source_time = 1/peak_freq

model = tr.tensor(data)

"""
source and receiver position --- x_s -> Tensor of shape [num_shots, num_sources_per_shot, num_dimensions]
--- x_r -> Tensor of shape [num_shots, num_receivers_per_shot, num_dimensions]
"""

x_r = tr.zeros(num_shots,num_receivers_per_shot,num_dims)
for j in range(num_receivers_per_shot):
  x_r[:,j,1] = coord_receivers[j] #receiver potition

x_s = tr.zeros(num_shots, num_sources_per_shot, num_dims)
for j in range(num_sources_per_shot):
  x_s[:,0,1] = tr.linspace(ymax/2, ymax, num_shots) #sources potition
  x_s[:,0,0] = 100#tr.linspace(0,(nz-1)*dx, len(x_s[:,0,1])) #depth

source_amplitudes = (dp.wavelets.ricker(peak_freq, nt, dt,
peak_source_time).reshape(-1, 1, 1).repeat(1, num_shots, num_sources_per_shot))

prop = dp.scalar.Propagator({'vp': model}, dx)
receiver_amplitudes = prop(source_amplitudes, x_s, x_r, dt)
```

**Comment by Ge0sz** [comment]
What was the input shape to print it like this? you used plt.imshow()? the shape of result is  [nt, num_shots, num_receivers_per_shot], yes?

**Comment by ar4** [comment]
I think you are asking what command I used to create the two plots in my previous message. I used `plt.imshow(receiver_amplitudes[...,0])` and `plt.imshow(receiver_amplitudes[...,1])` to produce them, plotting the data recorded by each of the two receivers.

**Comment by Ge0sz** [comment]
so in our example x is the shot of each source and y the time, yes?

**Comment by ar4** [comment]
Yes.

**Comment by Ge0sz** [comment]
also, instead of  x_s[:,0,0] = 100 for depth, can i use , x_s[:,0,0] = tr.linspace(0,(nz-1)*dx, len(x_s[:,0,1])) in order to have for each source different depth?

**Comment by ar4** [comment]
That should work, although note that the source will be moved to the nearest cell - if the cell size is 10x10m then a source located at (5, 5)m will be moved to (10,10)m, for example.

I am not sure what you are referring to when you ask about simultaneous shots. I can think of two possibilities:

1. The `num_sources_per_shot` refers to how many sources will be used in each shot. If you wish for each shot to only contain one source (so no simultaneous sources) then this should be set to 1. It is set to 1 in the example code above, so the data in `receiver_amplitudes[:,0,:]` will only involve the source located at `x_s[0,0,:]`, for example.
2. You may also be asking about whether parallel processing is used to perform the propagation in parallel. It is.

**Comment by Ge0sz** [comment]
Perfect, thank you!

**Comment by Ge0sz** [comment]
How can i increase the duration of the time of recording? by increasing 'nt'?

**Comment by ar4** [comment]
Yes.

**Comment by Ge0sz** [comment]
the coordinates -position of receiver and source should be in real distance or pixel?

**Comment by ar4** [comment]
The documentation says "The coordinates are specified in the same units as
dx and are with respect to the origin of the model." - i.e., real distance.
You can also see this in the examples, e.g., where the source coordinates
are set using x_s = torch.Tensor([[[0, 20 * dx]]])

**Comment by Ge0sz** [comment]
The result is the total displacement or there is a specific orientation for example horizontal-vertical? Thanks again :)

**Comment by ar4** [comment]
The package currently uses the scalar wave equation. I think the values can
therefore be best interpreted as being related to differential pressure.

**Comment by Ge0sz** [comment]
so it's acceleration?

**Comment by ar4** [comment]
Acceleration is usually a vector.

**Comment by Ge0sz** [comment]
pressure is also a vector

**Comment by ar4** [comment]
I think pressure is normally a scalar:
https://en.wikipedia.org/wiki/Pressure#Scalar_nature

**Comment by Ge0sz** [comment]
ok, can you suggest any way to calculate displacement?

**Comment by ar4** [comment]
I think displacement is inversely proportional to the spatial derivative of
pressure. If you need vertical displacement at a specific position, for
example, you could probably record the wavefield at a few adjacent cells in
the vertical direction around that position and use these to calculate the
vertical derivative there.

**Comment by ar4** [comment]
Not inversely proportional - I meant proportional with a negative coefficient.

**Comment by Ge0sz** [comment]
You mean: ΔP=-x , where ΔP the differential pressure and x the displacement? Also, how can i separate the vertical or horizontal component if the pressure is scalar?

**Comment by ar4** [comment]
By "differential pressure" I mean excess pressure - the difference from the
pressure in the unperturbed state. I believe displacement is proportional
to the negative spatial derivative of the differential pressure. If you
want vertical displacement, you calculate the vertical derivative of the
differential pressure, whereas if you want horizontal displacement then you
calculate the horizontal derivative of the differential pressure.

**Comment by ar4** [comment]
Did you get the result that you wanted?

**Comment by Ge0sz** [comment]
I wanted to ask, if its scalar and it depicts the total, how can calculate the displacement in a direction for example vertically?

**Comment by ar4** [comment]
I will try to clarify what I said:

With Deepwave, you can record the (differential/excess) pressure at
grid cells that you specify as receiver locations. I think the
displacement in a direction can be approximately calculated using the
negative spatial derivative of the pressure in that direction. If you
want the vertical displacement at the point (y,z), you can therefore
place receivers at (y,z-1) and (y,z+1) to record the pressure at those
two points and then use these recordings to calculate the vertical
derivative, and thus vertical displacement, there (vertical
displacement at (y,z) is approximately proportional to
p(y,z+1)-p(y,z-1)). The derivative can be calculated more accurately
by using a higher accuracy finite difference approximation (using more
receivers).

Does that make sense?

**Comment by Ge0sz** [comment]
Now yes! Thanks again
